subsys/nrf_security/src/drivers/cracen/cracenpsa/src/sign.c

```diff
@@ -112,7 +113,13 @@ static int cracen_signature_prepare_ec_prvkey(struct si_sig_privkey *privkey, ch
 		return status;
 	}
 
-	if (key_buffer_size != PSA_BITS_TO_BYTES(psa_get_key_bits(attributes))) {
+	if (PSA_KEY_LIFETIME_GET_LOCATION(psa_get_key_lifetime(attributes)) ==
+	    PSA_KEY_LOCATION_CRACEN_KMU) {
+		key_buffer = kmu_push_area;
+		key_buffer_size = sizeof(kmu_push_area);
+	}
+
+	if (key_buffer_size <= PSA_BITS_TO_BYTES(psa_get_key_bits(attributes))) {
```

Comment:
I think accepting oversize buffers here are wrong. (altough I guess it should have been validated in import/generate key anyways...?)

Reply:
I thought it should be fine as `key_buffer_size` is not used anymore in the function after this point